collector/collector.go

```diff
@@ -77,3 +77,28 @@ func registerCollector(collector string, isDefaultEnabled bool, factory factoryF
 func expandMetricName(prefix string, context ...string) string {
 	return strings.Join(append(context, prefix), `-`)
 }
+
+func newDesc(subsystem string, metric_name string, help_text string, labels []string) desc {
+	var name = prometheus.BuildFQName(namespace, subsystem, metric_name)
+	return desc{
+		name: name,
+		prometheus: prometheus.NewDesc(
+			name,
+			help_text,
+			labels,
+			nil,
+		),
+	}
+}
+
+func newMetric(metric_desc *desc, value float64, labels []string) metric {
+	return metric{
+		name: expandMetricName(metric_desc.name, labels...),
+		prometheus: prometheus.MustNewConstMetric(
+			metric_desc.prometheus,
+			prometheus.GaugeValue,
```

Comment:
Hmm ... I'd advise against making it an argument if we only ever need this one [kind](https://prometheus.io/docs/concepts/metric_types/) (this would unnecessarily increase the number of parameters at the call sites). But it could be renamed to e.g. `newGaugeMetric()` to prevent confusion.

Reply:
I ended up agreeing with you on this in my refactor.